src/google/protobuf/util/internal/utility.cc

```diff
@@ -302,7 +302,7 @@ bool IsMap(const google::protobuf::Field& field,
 string DoubleAsString(double value) {
   if (value == std::numeric_limits<double>::infinity()) return "Infinity";
   if (value == -std::numeric_limits<double>::infinity()) return "-Infinity";
-  if (::isnan(value)) return "NaN";
+  if (std::isnan(value)) return "NaN";
```

Comment:
From reading this thread:
http://stackoverflow.com/questions/570669/checking-if-a-double-or-float-is-nan-in-c
what I get is that "value != value" may not work reliable.

Could you elaborate why "MathLimits::IsNan(value)" doesn't work? To make the code portable we need a portable version of IsNan and I would prefer making the MathLimits one work for that purpose.